### PR TITLE
[build] Add skeleton for deployment of jars from Bintray to Sonatype

### PR DESCRIPTION
**Pull request by sgnn7**
Since Bintray is shutting down, this PR will move the publish target from them to
Sonatype.

AC-815

### CODE REVIEW

File: .gitlab-ci.yml
Comment (olivielpeau) [inline .gitlab-ci.yml:32]: should we explicitly use `set +x` when manipulating these secrets? (similar to what's done in the agent pipeline: https://github.com/DataDog/datadog-agent/blob/490b0bc2a07d7d6693d5d5955de4194e286a422c/.gitlab/package_build/rpm.yml#L20-L24)

File: .gitlab-ci.yml
Comment (olivielpeau) [inline .gitlab-ci.yml:14]: Not sure we want to restrict to the `master` branch? (I think we want any "stable" tag to trigger a release, whatever branch it's on?)

**Review by olivielpeau — APPROVED** [review]
Haven't reviewed all the details, but at a high level this LGTM, nice work 👍 

Just left a couple of questions/comments inline. Once answered, if this is confirmed to work well it looks ready for a merge

File: .gitlab-ci.yml
Comment (sgnn7) [reply .gitlab-ci.yml:14]: I guess maybe I was a bit concerned about security here - if someone makes a malicious branch without a review and then pushes a tag for that branch's commit, they can force a bad release. Since master is usually protected from direct pushes, this seemed like a slight mitigation for that and the manual trigger can be used for non-master builds. I can definitely remove this branch check if the tag triggers seem more reasonable for releasing.

File: .gitlab-ci.yml
Comment (sgnn7) [reply .gitlab-ci.yml]: Fixed

File: .gitlab-ci.yml
Comment (olivielpeau) [reply .gitlab-ci.yml:14]: ok, but overall at the moment anyone with write access to the repo can push directly to master as well, so not sure this mitigates much (?)

Maybe as a larger mitigation you could make this step _always_ manual, so that access to the internal gitlab is required to trigger releases?

(I don't have a strong opinion to be honest)

**Review by olivielpeau — DISMISSED** [review]
LGTM

**Review by olivielpeau — APPROVED** [review]
(travis build failed but probably a flake, worth a retry)

**Comment by sgnn7** [comment]
- Unable to trigger the same pipeline again in TravisCI due to `Oh no!
An error occurred. The build could not be restarted.` error. Using Travis CLI via GH oauth MFA did not work so I was unable to trigger the rebuild that way.
- [Manually triggered build](https://travis-ci.com/github/DataDog/jmxfetch/builds/221342218) for the commit succeeds

Since it seems like a flake and to prevent the need for another round of approvals I'll merge this and keep an eye on the master branch build.

**Comment by sgnn7** [comment]
[Master build of the merged commit was green](https://travis-ci.com/github/DataDog/jmxfetch/builds/221344524). Test failure was a flake.

**Pull request merged**
